### DISCUSSION TITLE
Upload data scheme and backend integration

### DISCUSSION THREAD

**Discussion by mrzdtydlntm** [TerriaJS/terriajs#5831] [Q&A] [answered]
Hello.
I want to ask some question about upload scheme in terriamap and adding database (using express). Before that, I want to make a converter for some dataset that didn't supported to read in terriamap and want to add it in upload button. When the uploaded dataset is on unsupported format, it will automatically converted. Where I can modify the upload scheme to add convert function?

Second, about connection with database using backend. Is there a guide to connect the terriamap with express?

Thank you for your reply.

Mirza

**Comment by hardreddata** [comment, marked as answer]
Hi @mrzdtydlntm 

I think most users who ultimately connect to a database would go through something like http://geoserver.org/ and serve the data as a WMS layer.

Someone else may be able to answer your question more directly.

Cheers.
